kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bytes-simplification.k

```diff
@@ -88,12 +88,11 @@ module BYTES-SIMPLIFICATION [symbolic]
       requires 0 <=Int W1 andBool 0 <=Int W2
       [simplification]
 
-    rule [buf-shift]:
-      #buf(W:Int, X <<Int Y) => #buf(W -Int (Y /Int 8), X) +Bytes #buf(Y /Int 8, 0)
-      requires 0 <=Int W andBool W <=Int 32
-       andBool 0 <=Int X andBool X <Int 2 ^Int (256 -Int Y)
-       andBool 0 <=Int Y andBool ( Y <Int W *Int 8 ) andBool Y modInt 8 ==Int 0
-      [simplification, preserves-definedness]
+    rule [shift-as-asWord]:
+      X <<Int Y => #asWord ( #buf ( 32 -Int (Y /Int 8) , X ) +Bytes #buf ( Y /Int 8 , 0 ) )
+      requires 0 <=Int X andBool X <Int 2 ^Int (256 -Int Y)
+       andBool 0 <=Int Y andBool Y <=Int 256 andBool Y modInt 8 ==Int 0
+      [simplification, concrete(Y), preserves-definedness]
```

Comment:
Is it possible that this might get in the way of the other `X <<Int Y` rules, like `(X <<Int Y) <Int pow256 => true` above? If so, should we lower this rule's priority?

Reply:
Alright, if we haven't seen any issues and the expressions will still simplify anyway then it should be fine.